numpy/core/code_generators/generate_umath.py

```diff
@@ -276,7 +293,7 @@ def english_upper(s):
     Ufunc(2, 1, Zero,
           docstrings.get('numpy.core.umath.add'),
           'PyUFunc_AdditionTypeResolver',
-          TD(notimes_or_obj, simd=[('avx2', ints)]),
+          TD(notimes_or_obj, dispatch=[('loops_fast', ints)]),
```

Comment:
Is there a case where dispatch will contain something other than `[('loops_fast', ints)]` ?

Reply:
if the optimizations aren't disabled by `--disable-optimization`, `compiler_opt::try_dispatch` will generate a separated config file for each source that ends with ".dispatch.c" depending on the specified CPU features that defined on the top of source file,  e.g. "/*@ targets SSE41 AVX2 VSX VSX2*/".
The dispatched config header (DCH) taking the same name of the dispatch source file but of course the extension going to be ".h" , also this config file provides two special macros `NPY__CPU_DISPATCH_CALL` and `NPY__CPU_DISPATCH_BASELINE_CALL` going to use later by the defined macros in `npy_cpu_dispatch.h` to determine which CPU features that will need to dispatch in runtime.

>  Is there a case where dispatch will contain something other than [('loops_fast', ints)] ?

So yeah definitely it will. for example, assume we going to write  SIMD code to improve some operations called `XX` for integer types and we going to put this code in a separate C source file named `xx.dispatch.c` then our DCH going to be `xx.dispatch.h`, so the TD properties must contains "dispatch=[('`xx`', ints)])".